sdk/core/core-http/src/fetchHttpClient.ts

```diff
@@ -195,9 +196,33 @@ export abstract class FetchHttpClient implements HttpClient {
 
       throw fetchError;
     } finally {
-      // clean up event listener
-      if (httpRequest.abortSignal && abortListener) {
-        httpRequest.abortSignal.removeEventListener("abort", abortListener);
+      // clean up event listener. for streaming operations clean up on close/end/error event.
+      const cleanUpStreamListener = (stream: NodeJS.ReadableStream) => {
+        const cleanupCallback = () => {
+          stream.removeListener("close", cleanupCallback);
+          stream.removeListener("end", cleanupCallback);
+          stream.removeListener("error", cleanupCallback);
+          if (abortListener) {
+            httpRequest.abortSignal?.removeEventListener("abort", abortListener);
```

Comment:
So this cleans up the listener - but will cancelation work as expected? Does `node-fetch` actually stop upload / download streams when the signal is given?

I believe that it could, I just want to make sure we've actually tested it.

Reply:
Yes I tested it using the storage repo code.  node-fetch will destroy request body and emit error to the response body when handling abort event https://github.com/node-fetch/node-fetch/blob/v2.6.1/src/index.js#L60